org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/handler/ZigBeeThingHandler.java

```diff
@@ -423,6 +439,12 @@ private synchronized void doNodeInitialisation() {
         coordinatorHandler.serializeNetwork();
     }
 
+    private synchronized void reinitializeDevice() {
+        logger.debug("{}: Reinitializing device", nodeIeeeAddress);
+        dispose();
+        initialize();
```

Comment:
This seems a bit strange - maybe this is also Hennings issue.

This method ```reinitializeDevice``` doesn't reinitialise the device - it reinitialises the thing handler and the device. I would have expected that when the user wants to reinitialise the device, it just does that.

Or, do you think that because we have updated the device, that we also need to reinitialise all the converters? I'm not sure this is the case (I'd need to think it through some more - maybe you've already done that?). The clusters won't change, so I don't think reinitialising the converters and redetecting the channels and polling is really necessary?

I could be wrong though, and maybe a full reinitialisation and repoll etc is really needed, but this wasn't my expectation when I first introduced this concept.

Reply:
The idea was to re-use the existing methods and so rely on the existing initialization. However, I understand your objection and so I took a closer look at the initialization part. As far as I can see we can initialize the device without the need to re-initialize anything else. I'm implementing a method that initializes only the devices. PR will follow soon.